### PR TITLE
moved data files to data directory, also added use of TEST_FILE_DIR and improved CI system

### LINKED ISSUE #331: improve CI system
* Create a developer build which does FTP testing, address santizer, doxygen build, warning check, code coverage, and Debug build. Turn these settings off in all other builds. Use develop branch of all required NCEPLIBS libraries and the latest version of jasper.
* Create a linux_versions workflow that checks with different versions of dependencies.
* Create a linux_settings workflow that tests different build options with the most recent versions of dependencies.
* Use ubuntu-latest/macos-latest everywhere.
* Use caches everywhere they can be used, including on code checkouts.

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #329 
Fixes #331

Improving testing of files fetched from FTP site.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
These tests will continue to improve incrementally, but this is a good place to merge these changes.

@Hang-Lei-NOAA and @AlexanderRichert-NOAA please inspect.

File: tests/test_putgb2.F90 [hunk 1]
```diff
@@ -83,11 +83,11 @@ program test_putgb2
...
-  call putgb2(2, gfld, iret)
-  if (iret .ne. 0) then
-     print *, 'putgb2 failed with iret value: ', iret
-     stop 7
-  end if
+  ! call putgb2(2, gfld, iret)
+  ! if (iret .ne. 0) then
+  !    print *, 'putgb2 failed with iret value: ', iret
+  !    stop 7
+  ! end if
```
Comment (AlexanderRichert-NOAA) [inline tests/test_putgb2.F90:90]: What are these lines being commented for?

**Comment by AlexanderRichert-NOAA** [comment]
gdaswave.t00z.wcoast.0p16.f000.grib2.idx is still under tests/ rather than tests/data/, is that intentional?

File: tests/test_putgb2.F90 (hunk 1 above)
Comment (edwardhartnett) [reply tests/test_putgb2.F90:90]: Good catch! I will circle around and check...

**Review by Hang-Lei-NOAA — APPROVED** [review]

**Comment by edwardhartnett** [comment]
I am breaking this work up into smaller pieces, which I'll submit in separate PRs.

**Comment by edwardhartnett** [comment]
Too many problems in this PR. I will close and circle around again...

**Pull request closed without merging**